docs/tutorials/channels.md

```diff
@@ -305,344 +306,673 @@ publish = false
+    /// Get the maximum height of the decided values in the store
+    pub async fn max_decided_value_height(&self) -> Option<Height>
 
-We can simply respond by telling the engine to start consensus at the current height,
-which is initially one.
+    /// Get the decided value at the given height
+    pub async fn get_decided_value(
+        &self,
+        height: Height,
+    ) -> Result<Option<DecidedValue>, StoreError>
 
-```rust
-            AppMsg::ConsensusReady { reply } => {
-                info!("Consensus is ready");
+    /// Store a decided value
+    pub async fn store_decided_value(
+        &self,
+        certificate: &CommitCertificate<TestContext>,
+        value: Value,
+    ) -> Result<(), StoreError>
 
-                if reply
-                    .send(ConsensusMsg::StartHeight(
-                        state.current_height,
-                        genesis.validator_set.clone(),
-                    ))
-                    .is_err()
-                {
-                    error!("Failed to send ConsensusReady reply");
-                }
-            }
-```
+    /// Store an undecided proposal
+    pub async fn store_undecided_proposal(
+        &self,
+        value: ProposedValue<TestContext>,
+    ) -> Result<(), StoreError>
 
-The next message to handle is the `StartRound` message, signaling to the app
-that consensus has entered a new round (including the initial round 0).
+    /// Get the undecided proposal at the given height and round
+    pub async fn get_undecided_proposal(
+        &self,
+        height: Height,
+        round: Round,
+    ) -> Result<Option<ProposedValue<TestContext>>, StoreError>
 
-We can use that opportunity to update our internal state:
+    /// Prune the store, removing all decided values below the given height
+    pub async fn prune(&self, retain_height: Height) -> Result<Vec<Height>, StoreError>
 
-```rust
-            AppMsg::StartedRound {
-                height,
-                round,
-                proposer,
-            } => {
-                info!(%height, %round, %proposer, "Started round");
+    /// Remove undecided proposals matching the given value id
+    pub async fn remove_undecided_proposals_by_value_id(
+        &self,
+        value_id: ValueId,
+    ) -> Result<(), StoreError>
 
-                state.current_height = height;
-                state.current_round = round;
-                state.current_proposer = Some(proposer);
-            }
+    /// Get the undecided proposal matching the given value id
+    pub async fn get_undecided_proposal_by_value_id(
+        &self,
+        value_id: ValueId,
+    ) -> Result<Option<ProposedValue<TestContext>>, StoreError>
 ```
 
-At some point, we may end up being the proposer for that round, and the engine
-will then ask us for a value to propose to the other validators.
+Note that the implementation of this store is up to the application developer. One could have used a simple in-memory store.
```

Comment:
The store must be persistent, otherwise the application cannot be restarted in some cases.

```suggestion
Note that the implementation of this store is up to the application developer, who may choose a different underlying database such as RocksDB, LevelDB, or [Fjall](https://github.com/fjall-rs/fjall).
```